redis-lettuce/src/main/java/io/micronaut/configuration/lettuce/cache/RedisCache.java

```diff
@@ -207,6 +215,46 @@ public String getName() {
         return value;
     }
 
+    public <K> Map<K, Object> get(Collection<K> keys) {
+      return get(keys, Argument.of(Object.class));
+    }
+
+    public <K, T> Map<K, T> get(Collection<K> keys, Argument<T> requiredType) {
+        Map<byte[], K> serializedKeyMap = new HashMap<>(keys.size());
+        byte[][] serializedKeys = new byte[keys.size()][];
+        int i = 0;
+        for (K key: keys) {
+            final byte[] serializedKey = this.serializeKey(key);
+            serializedKeys[i++] = serializedKey;
+            serializedKeyMap.put(serializedKey, key);
+        };
+
+        List<KeyValue<byte[], byte[]>> data = redisStringCommands.mget(serializedKeys);
+        final Map<K, T> results;
+        if (data != null) {
+            results = new LinkedHashMap<>(data.size());
+            for (KeyValue<byte[], byte[]> result: data) {
+                K originalKey = serializedKeyMap.get(result.getKey());
+                if (result.hasValue()) {
+                  final T deserialized = valueSerializer.deserialize(result.getValue(), requiredType)
+                      .orElseThrow(() ->
+                          new ConversionErrorException(requiredType,
+                              new IllegalArgumentException("Cannot convert cached value for [" + originalKey + "] to target type: " + requiredType.getType() + ". Considering defining a TypeConverter bean to handle this case.")));
+                  results.put(originalKey, deserialized);
+                  if (expireAfterAccess != null) {
+                      redisKeyCommands.pexpire(result.getKey(), expireAfterAccess);
+                  }
+                } else {
+                  results.put(originalKey, null);
+                }
+            }
+        } else {
+            results = new LinkedHashMap<>();
```

Comment:
Can this not be `Collections.emptyMap()`? And should we not wrap the result in `Collections.unmodifiableMap`?

Reply:
We could, though we're not doing anything with the collection afterwards, we're just handing it off to the caller.  I wouldn't _expect_ someone to modify the collection, but it also doesn't seem like there would be any harm if they did so.